cobc/cobc.c

```diff
@@ -8937,6 +8946,8 @@ finish_setup_compiler_env (void)
 
 	if (getenv ("COBC_GEN_DUMP_COMMENTS")) {
 		cb_wants_dump_comments = 1;
+		/* Disable "new" symbol table if requesting dump comments */
+		cb_flag_symbols = 0;
```

Comment:
please leave `cb_wants_dump_comments ` outside of that; if it is set then the old dump code should be generated - but only as comments (this should already be in its code)
please add a separate `if (getenv ("COBC_GEN_DUMP_AS_CODE")) {cb_flag_symbols = 0;]` here, so we can enable it if necessary - and until 4 also to test the backward compatibility of libcob.